tests/rest/admin/test_admin.py

```diff
@@ -337,7 +337,7 @@ def test_purge_room(self):
             "local_invites",
             "room_account_data",
             "room_tags",
-            "state_groups",
+            # "state_groups",  # Current impl leaves orphaned state groups around.
```

Comment:
Is this the likely cause of https://github.com/matrix-org/synapse/issues/6566?

Reply:
Ah, yup, as we don't delete by `room_id` on `state_groups` but indirect via the `events` table. That coupled with the fact that we now are more proactive with creating state groups that aren't referenced by events means we do end up with some orphaned bits and pieces.

Reply:
> Ah, yup, as we don't delete by room_id on state_groups

Why not?

Reply:
Mainly so that we could avoid adding another index to `state_groups`, but I think we might just want to do that anyway.